src/rewriter/mod.rs

```diff
@@ -266,22 +261,29 @@ impl<'h, O: OutputSink, H: HandlerTypes> Debug for HtmlRewriter<'h, O, H> {
 fn handler_adjust_charset_on_meta_tag<'h, H: HandlerTypes>(
     encoding: SharedEncoding,
 ) -> (Cow<'h, crate::Selector>, ElementContentHandlers<'h, H>) {
+    let mut set_once = false;
```

Comment:
Nit: let's rename it to `found` and also add a blank line after the variable declaration